modules/rational-compile.el

```diff
@@ -21,28 +21,59 @@
 
 ;;; Code:
 
+
 ;;;; Variables:
-;; We define a variable to enable/disable modules autocompilation.
+;; We define a variable to enable/disable modules compilation.
 (defvar rational-compile-modules t
-  "When non-nil, autocompile emacs lisp sources for the modules on startup.")
+  "When non-nil, compile emacs lisp sources for the modules on startup.")
+
+;; We define a variable to enable/disable user's modules compilation.
+(defvar rational-compile-user-modules t
+  "When non-nil, compile user's emacs lisp sources for the modules on startup.")
+
+(defvar rational-compile-module-list '()
+  "A list of modules to compile.")
+
+(defvar rational-compile-modules-path
+  `(,(expand-file-name "modules" rational-config-path)
+    ,(expand-file-name "modules" user-emacs-directory))
+  "Path where to locate modules.")
 
 (defvar rational-compile-user-configuration t
-  "When non-nil, autocompile emacs lisp sources for the modules on startup.")
+  "When non-nil, compile emacs lisp sources for the modules on startup.")
 
 (defvar rational-compile-init-files nil
-  "When non-nil, autocompile init files on startup.")
+  "When non-nil, compile init files on startup.")
 
 (defvar rational-compile-on-save nil
-  "When non-nil, autocompile the file after saving it.")
+  "When non-nil, compile the file after saving it.")
 
 (defvar rational-compile-extra-directories-list nil
-  "List of extra directories to autocompile.")
+  "List of extra directories to compile.")
 
 (defvar rational-compile-init-files-list
   '("early-init.el" "init.el")
   "List of initialization file names.")
 
+(defvar rational-compile-config-files-list
+  '("early-config.el" "config.el")
+  "List of configuration file names.")
+
+
 ;;;; Functions:
+;; A function to locate a module's source file:
+(defun rational-locate-module-file (module &optional path)
+  (setq-local module-path nil)
```

Comment:
maybe this could be simplified to just:
```emacs-lisp
(seq-find (lambda (dir) (file-exists-p (expand-file-name (format "%s.el" (symbol-name module)) dir)))
          (or path (rational-compile--config-dirs-list)))
```

Reply:
That's great! I didn't remember about this function (`seq-find`); haven't heard of it in years and I totally forgot it.